Content.Server/GameObjects/Components/Atmos/Piping/PipeNetDeviceComponent.cs

```diff
@@ -27,6 +29,12 @@ public override void OnRemove()
             LeaveGridAtmos();
         }
 
+        public void Update()
+        {
+            var message = new PipeNetUpdateMessage();
+            SendMessage(message);
```

Comment:
Creating the same empty message per pipe net device every update tick is expensive. Try caching the empty message and sending it over and over as needed, or use interfaces to update the rest of components instead.